core/types.rs

```diff
@@ -780,17 +682,32 @@ impl PartialEq<Value> for Value {
     }
 }
 
+fn int_float_cmp(int: i64, float: f64) -> std::cmp::Ordering {
+    if float.is_nan() {
+        return std::cmp::Ordering::Greater;
+    }
+
+    if float < -9223372036854775808.0 {
+        return std::cmp::Ordering::Greater;
+    }
+
+    if float >= 9223372036854775808.0 {
+        return std::cmp::Ordering::Less;
+    }
```

Comment:
could you add comments to this, I imagine this is related to some precision stuff with floats.